src/nvim/ex_cmds2.c

```diff
@@ -2788,6 +2788,43 @@ static FILE *fopen_noinh_readbin(char *filename)
   return fdopen(fd_tmp, READBIN);
 }
 
+typedef struct {
+  char_u *buf;
+  uint32_t pointer;
+} GetStrLineCookie;
+
+static char_u *get_str_line(int c, void *cookie, int ident)
+{
+  GetStrLineCookie *p = (GetStrLineCookie *)cookie;
+  uint32_t i = p->pointer;
+  if (strlen((char *)p->buf) <= p->pointer) {
+    return NULL;
+  }
+  while (!(p->buf[i] == '\n' || p->buf[i] == '\0')) {
+    i++;
+  }
+  char buf[2046];
+  char *dst;
+  dst = xstpncpy(buf, (char *)p->buf+p->pointer, i - p->pointer);
+  if ((uint32_t)(dst - buf) != i - p->pointer) {
+    smsg(_("nvim_source error parsing command %s"), p->buf);
+  }
+  buf[i-p->pointer]='\0';
```

Comment:
isn't this `buf[0] = '\0'` ?

Reply:
We have a loop where we increment i at lines 2803-2805